File: src/test/java/com/example/scaffold/PingTest.java
package com.example.scaffold;

import com.example.scaffold.frameworks.test.web.IntegrationTest;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static com.example.scaffold.frameworks.test.web.PathVariables.variables;
import static org.hamcrest.Matchers.is;

public class PingTest extends IntegrationTest {
    @Test
    void should_reply_pong() {
        post("/ping", variables(Map.of()), Map.of("data", "pong"), null).statusCode(is(200)).body("data", is("pong"));
    }
}

File: src/test/kotlin/com/qomolangma/infra/PingTest.kt
package com.qomolangma.infra

import com.qomolangma.ApiTest
import com.qomolangma.frameworks.test.web.Documentation
import com.qomolangma.frameworks.test.web.PathVariables.Companion.variables
import com.qomolangma.frameworks.test.web.WithGrantedAuthorities
import org.hamcrest.Matchers.`is`
import org.junit.jupiter.api.Test
import org.springframework.restdocs.payload.PayloadDocumentation.*

class PingTest : ApiTest() {
    @Test
    @WithGrantedAuthorities(["grantedAuthority"])
    fun should_reply_pong() {
        val post = post("/ping", variables(HashMap<String, Any>()), mapOf(Pair("data", "pong")), document())
        post.statusCode(`is`(200)).body("data.ping", `is`("pong")).body("data.message", `is`("ok"))
    }

    override fun document(): Documentation {
        return Documentation.doc(
            "test.ping", requestFields(
                fieldWithPath("data").description("Test data")
            ), responseFields(
                fieldWithPath("code").description("response code"),
                fieldWithPath("msg").description("response msg"),
                fieldWithPath("data").description("response data"),
                fieldWithPath("data.ping").ignored(),
                fieldWithPath("data.message").ignored()
            )
        )
    }
}
